Sources/DolbyIORTSCore/StreamCoordinator.swift

```diff
@@ -54,49 +50,83 @@ open class StreamCoordinator {
         self.subscriptionManager.delegate = self
 
         startStateObservation()
+        startStateMachineTasksSerialExecutor()
     }
-    #endif
 
     private func startStateObservation() {
         Task {
             await stateMachine.statePublisher
                 .sink { [weak self] state in
                     guard let self = self else { return }
+                    
+                    switch state {
+                    case let .error(state):
+                        switch state.error {
+                        case .connectFailed:
+                            self.scheduleReconnection()
+                        default:
+                            //No-op
+                            break
+                        }
+                    case .stopped:
+                        self.scheduleReconnection()
+                    default:
+                        // No-op
+                        break
+                    }
 
                     // Populate updates public facing states
                     self.stateSubject.send(StreamState(state: state))
                 }
             .store(in: &subscriptions)
         }
     }
-
-    static func setStreamCoordinatorConfiguration(_ configuration: StreamCoordinatorConfiguration) {
-        Self.configuration = configuration
+    
+    private func scheduleReconnection() {
+        taskScheduler.scheduleTask(timeInterval: Defaults.retryConnectionTimeInterval) { [weak self] in
+            guard let self = self, let streamDetail = self.activeStreamDetail else { return }
+            Task {
+                self.taskScheduler.invalidate()
+                _ = await self.connect(
+                    streamName: streamDetail.streamName,
+                    accountID: streamDetail.accountID
+                )
+            }
+        }
+    }
+    
+    private func startStateMachineTasksSerialExecutor() {
+        Task {
+            let tasksStreams = AsyncStream<CoordinatorTask> { continuation in
```

Comment:
Either `streamTasks` or `streamsTasks` seem to be better

Reply:
To make it sound like a type of AsyncStream - I reckon it's good to name it `....Stream`. I see it does not need to be Plural though. Fixing it now